addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -187,7 +188,12 @@ def write(self, vals):
         with self._error_checking(skip=skip_check, employee_ids=employee_ids):
             return super(HrWorkEntry, self).write(vals)
 
+    def _error_raising(self):
+        if any(w.state == 'validated' for w in self):
+            raise UserError(_("This work entry is used in a validated payslip. You can't delete it."))
```

Comment:
I am wondering, whether this check should be here on in payroll.. because just in hr_work_entry, you don't have payroll.

Reply:
Nevermind, it should be here